lib/cocoapods/installer/analyzer/target_inspector.rb

```diff
@@ -0,0 +1,196 @@
+
+          result = TargetInspectionResult.new
+          result.target_definition = target_definition
+          result.project_path = project_path
+          result.project_target_uuids = targets.map(&:uuid)
+          result.build_configurations = compute_build_configurations(targets)
+          result.platform = compute_platform(targets)
+          result.archs = compute_archs(targets)
+          result
+        end
+
+        #-----------------------------------------------------------------------#
+
+        private
+
+        # Returns the path of the user project that the #target_definition
+        # should integrate.
+        #
+        # @raise  If the project is implicit and there are multiple projects.
+        #
+        # @raise  If the path doesn't exits.
+        #
+        # @return [Pathname] the path of the user project.
+        #
+        def compute_project_path
+          if target_definition.user_project_path
+            path = config.installation_root + target_definition.user_project_path
+            path = "#{path}.xcodeproj" unless File.extname(path) == '.xcodeproj'
+            path = Pathname.new(path)
+            unless path.exist?
+              raise Informative, 'Unable to find the Xcode project ' \
+              "`#{path}` for the target `#{target_definition.label}`."
+            end
+          else
+            xcodeprojs = config.installation_root.children.select { |e| e.fnmatch('*.xcodeproj') }
+            if xcodeprojs.size == 1
+              path = xcodeprojs.first
+            else
+              raise Informative, 'Could not automatically select an Xcode project. ' \
+              "Specify one in your Podfile like so:\n\n" \
+              "    xcodeproj 'path/to/Project.xcodeproj'\n"
+            end
+          end
+          path
+        end
+
+        # Returns a list of the targets from the project of #target_definition
+        # that needs to be integrated.
+        #
+        # @note   The method first looks if there is a target specified with
+        #         the `link_with` option of the {TargetDefinition}. Otherwise
+        #         it looks for the target that has the same name of the target
+        #         definition.  Finally if no target was found the first
+        #         encountered target is returned (it is assumed to be the one
+        #         to integrate in simple projects).
+        #
+        # @param  [Xcodeproj::Project] user_project
+        #         the user project
+        #
+        # @return [Array<PBXNativeTarget>]
+        #
+        def compute_targets(user_project)
+          native_targets = user_project.native_targets
+          if link_with = target_definition.link_with
+            targets = native_targets.select { |t| link_with.include?(t.name) }
+            raise Informative, "Unable to find the targets named `#{link_with.to_sentence}` to link with target definition `#{target_definition.name}`" if targets.empty?
```

Comment:
`link_with.to_sentence` shouldn't be inside backticks, since it would put the whole sentence inside the one pair